api/grpcserver/admin_service.go

```diff
@@ -48,7 +48,11 @@ func (a AdminService) CheckpointStream(req *pb.CheckpointStreamRequest, stream p
 	// - on the client side (default limit on the receiving end)
 	// - locally as the node already loads db query result in memory
 	snapshot := types.LayerID(req.SnapshotLayer)
-	err := checkpoint.Generate(stream.Context(), afero.NewOsFs(), a.db, a.dataDir, snapshot)
+	numAtxs := int(req.NumAtxs)
+	if numAtxs < 2 {
+		numAtxs = 2 // default
+	}
+	err := checkpoint.Generate(stream.Context(), afero.NewOsFs(), a.db, a.dataDir, snapshot, numAtxs)
```

Comment:
i think NumAtxs should be a node config, not an rpc argument. the goal is that the whole network should generate the same snapshot.

Reply:
just read https://github.com/spacemeshos/go-spacemesh/pull/4617#issuecomment-1614551619

my argument is that node operators should be able to generate the snapshot themselves to verify against spacemesh published version. so nodes have to agree on the default number. 2 above is too small. for mainnet i thought we agreed on 4. 

the default constant should be defined as a constant as well.

Reply:
@dshulyak 
considering the following:
- different network can have different sensible defaults (maybe 4 for mainnet, 10 for testnet)
- we want the node operators to be able to validate our snapshot easily
i feel it's simpler to put it as a node configuration.

however, for spacemesh operator, it would be simpler if it's an RPC argument. maybe we should have an RPC to update the default number of ATXs for snapshot instead, and leave the Checkpoint RPC unchanged?

Reply:
i have a hard time rationalizing making it a rpc req argument.
two purposes for the Checkpoint RPC:
A. spacemesh operator to generate the snapshot
B. node operator to validate the snapshot

for A, there is currently no way for spacemesh operators to know that numAtxs==4 is not enough. therefore they need to generate another snapshot where numAtxs is 5. and by the time we realize the number of snapshot is not enough, the damage is already done. so allowing as argument doesn't make sense to me.

the key for me here is B and A need to generate the same data. so they should use the same default value.
i still think a config value makes more sense.

Reply:
The way I implemented it now:

- `numAtxs` is a second parameter for the request that specifies how many ATXs per identity should be included in the checkpoint
- the checkpoint now contains a "command" field that shows the command that was used to generate it.

Reply:
thanks. that works for me.